quasar-core/src/main/java/co/paralleluniverse/strands/channels/disruptor/StrandBlockingWaitStrategy.java

```diff
@@ -57,11 +53,13 @@ public long waitFor1(long sequence, Sequence cursorSequence, Sequence dependentS
             try {
                 for (int i = 0; (availableSequence = cursorSequence.get()) < sequence; i++) {
                     barrier.checkAlert();
-                    processorNotifyCondition.await(i);
+                    processorNotifyCondition.await(i, 1, TimeUnit.NANOSECONDS);
```

Comment:
When we start running fiber consumers they never end they always wait for barrier to release for new sequences and If we block the fiber which is a fork join task behind the scene, forkjoinpool doesn't take another fork join task for executing so everything stucking there. Say you have 3 thread running in thread pool with id 1,2,3 In this case they always execute fibers with id 1,2,3 I couldn't find another way to execute other fibers rather than solution up there Please help me on this